pyserini/trectools/_base.py

```diff
@@ -90,36 +116,45 @@ def get_all_topics_from_runs(runs) -> Set[str]:
         return all_topics
 
     @staticmethod
-    def merge(runs, aggregation: str, depth: int = None, k: int = None):
+    def merge(runs, aggregation: AggregationMethod, depth: int = None, k: int = None):
         if len(runs) < 2:
             raise Exception('Merge requires at least 2 runs.')
 
         rows = []
 
-        if aggregation == 'sum':
+        if aggregation == AggregationMethod.SUM:
             for topic in TrecRun.get_all_topics_from_runs(runs):
                 doc_scores = dict()
                 for run in runs:
                     for topic, _, docid, _, score, _ in run.get_docs_by_topic(topic, depth).to_numpy():
                         doc_scores[docid] = doc_scores.get(docid, 0.0) + score
 
                 sorted_doc_scores = sorted(iter(doc_scores.items()), key=lambda x: (-x[1], x[0]))
-                sorted_doc_scores = sorted_doc_scores if k is None else sorted_doc_scores[:k] 
+                sorted_doc_scores = sorted_doc_scores if k is None else sorted_doc_scores[:k]
 
                 for rank, (docid, score) in enumerate(sorted_doc_scores, start=1):
                     rows.append((topic, 'Q0', docid, rank, score, 'merge_sum'))
 
         else:
-            raise Exception(f'Unknown aggregation method {aggregation} detected.')
+            raise NotImplementedError()
 
-        return TrecRun.get_trec_run_by_list(rows)
+        return TrecRun.get_trec_run_from_list(rows)
 
     @staticmethod
-    def get_trec_run_by_list(rows, run=None):
+    def get_trec_run_from_list(rows, run: TrecRun = None) -> TrecRun:
```

Comment:
what type is `rows`? not immediately obvious what this method is doing?

Reply:
slow. The faster way is to dump df to a Numpy array, perform all the necessary operations, and append results to a list.

```rows.append((topic, 'Q0', docid, rank, 1 / (rrf_k + rank), tag))```


With the variable `rows`, we can this function to convert it to a TrecRun.

If `run` is given, then we update the given TrecRun with `rows`.

Reply:
Added proper doc string to explain what this function is doing.